src/main/java/edu/wpi/first/shuffleboard/WidgetPaneController.java

```diff
@@ -241,4 +303,104 @@ private Menu createChangeMenus(WidgetTile tile) {
+        layout.size.getWidth(), layout.size.getHeight(), n -> tile == n) && previousColumn >= 0) {
+      // Great, we can move it
+      return Optional.of(() -> GridPane.setColumnIndex(tile, previousColumn));
+    } else if (layout.origin.col > 0) {
+      // Check to see if we can move any tiles to the left, if possible
+      List<Runnable> runs = IntStream.range(layout.origin.row, layout.origin.row + layout.size.getHeight())
+          .mapToObj(row -> pane.tileAt(previousColumn, row))
+          .filter(Optional::isPresent) // guaranteed to be at least one tile
+          .map(Optional::get)
+          .distinct() // need to make sure we have no repeats, or n-row tiles will get moved n times
+          .filter(t -> tile != t)
+          .map(this::moveTileLeft) // recursion here
+          .filter(Optional::isPresent)
+          .map(Optional::get)
+          .collect(Collectors.toList());
+      if (runs.isEmpty()) {
+        // Can't move or shrink any tiles to the left of this one to make it move
+        return tryShrinkTile(tile, t -> t.getSize().getWidth(), t -> new TileSize(t.getWidth() - 1, t.getHeight()));
+      } else {
+        return Optional.of(() -> {
+          runs.forEach(Runnable::run);
+          GridPane.setColumnIndex(tile, previousColumn);
+        });
+      }
+    } else {
+      return tryShrinkTile(tile, t -> t.getSize().getWidth(), t -> new TileSize(t.getWidth() - 1, t.getHeight()));
+    }
+  }
+
+  private void moveTileUp(WidgetTile tile, int count) {
+    for (int i = 0; i < count; i++) {
+      Optional<Runnable> move = moveTileUp(tile);
+      if (move.isPresent()) {
+        move.get().run();
+      } else {
+        // Can't move any further, bail
+        break;
+      }
+    }
+  }
+
+  private Optional<Runnable> moveTileUp(WidgetTile tile) {
+    TileLayout layout = pane.getTileLayout(tile);
+    final int previousRow = layout.origin.row - 1;
+    if (!pane.isOverlapping(layout.origin.col, previousRow,
+        layout.size.getWidth(), layout.size.getHeight(), n -> tile == n) && previousRow >= 0) {
+      // Great, we can move it
+      return Optional.of(() -> GridPane.setRowIndex(tile, previousRow));
+    } else if (layout.origin.row > 0) {
+      // Check to see if we can move any tiles up, if possible
+      List<Runnable> runs = IntStream.range(layout.origin.col, layout.origin.col + layout.size.getWidth())
+          .mapToObj(col -> pane.tileAt(col, previousRow))
+          .filter(Optional::isPresent) // guaranteed to be at least one tile
+          .map(Optional::get)
+          .distinct() // need to make sure we have no repeats, or n-row tiles will get moved n times
+          .filter(t -> tile != t)
+          .map(this::moveTileUp) // recursion here
+          .filter(Optional::isPresent)
+          .map(Optional::get)
+          .collect(Collectors.toList());
+      if (runs.isEmpty()) {
+        // Can't move or shrink any tiles to the left of this one to make it move
```

Comment:
The preference here is weird. We should probably try shrinking the **current** tile before we shrink ones to the left of it, because that seems more intuitive.